### PR TITLE
[back/front] Improve relations based dashboard widgets (#2921)

### LINKED ISSUE #2921: [back/front] Improve relations based dashboard widgets
## Use case

Improve relations based dashboard widgets
- Prevent dashbard complete failure when a widget fail (each widget is error independant)
- Add some colors and information to have a better idea of what is configured
- Improve some existing use cases that generate backend errors
- Introduce the "contains" inner relationships to allow more widget use cases.
- Convert workspaceHeader to pure function
- Improve backend aggregation strategy

### LINKED ISSUE #2963: Specific dashboard filters cause crash of the dashboard
## Description

When adding entities visualization into a dashboard and choosing filters for entity type and related entity type, the dashboard will crash due to invalid GraphQL query.

## Environment

1. OpenCTI version: 5.5.3
2. OpenCTI client: frontend

## Reproducible Steps

Steps to create the smallest reproducible scenario:
1. Create a new dashboard and add a visualization:
image

2. Add a filter for entity type: malware, related entity: specific intrusion-set:
image

4. After validation and creation, the dashboard will crash with error:
`
"{\"category\":\"APP\",\"error\":{\"stacktrace\":[\"GraphQLError: Variable \\\"$timeSeriesParameters\\\" got invalid value \\\"relationship_type\\\" at \\\"timeSeriesParameters[0].filters[0].key\\\"; Value \\\"relationship_type\\\" does not exist in \\\"StixCoreObjectsFilter\\\" enum.\",\"at onError (/opt/opencti/build/node_modules/graphql/execution/values.js:147:11)\",\"at coerceInputValueImpl (/opt/opencti/build/node_modules/graphql/utilities/coerceInputValue.js:154:9)\",\"at coerceInputValueImpl (/opt/opencti/build/node_modules/graphql/utilities/coerceInputValue.js:49:14)\",\"at coerceInputValueImpl (/opt/opencti/build/node_modules/graphql/utilities/coerceInputValue.js:79:13)\",\"at coerceInputValueImpl (/opt/opencti/build/node_modules/graphql/utilities/coerceInputValue.js:49:14)\",\"at coerceInputValueImpl (/opt/opencti/build/node_modules/graphql/utilities/coerceInputValue.js:117:34)\",\"at /opt/opencti/build/node_modules/graphql/utilities/coerceInputValue.js:75:16\",\"at Function.from (<anonymous>)\",\"at coerceInputValueImpl (/opt/opencti/build/node_modules/graphql/utilities/coerceInputValue.js:73:20)\",\"at coerceInputValueImpl (/opt/opencti/build/node_modules/graphql/utilities/coerceInputValue.js:117:34)\",\"at /opt/opencti/build/node_modules/graphql/utilities/coerceInputValue.js:75:16\",\"at Function.from (<anonymous>)\",\"at coerceInputValueImpl (/opt/opencti/build/node_modules/graphql/utilities/coerceInputValue.js:73:20)\",\"at coerceInputValue (/opt/opencti/build/node_modules/graphql/utilities/coerceInputValue.js:32:10)\",\"at coerceVariableValues (/opt/opencti/build/node_modules/graphql/execution/values.js:132:52)\",\"at getVariableValues (/opt/opencti/build/node_modules/graphql/execution/values.js:45:21)\"]},\"inner_relation_creation\":0,\"level\":\"error\",\"message\":\"API Call\",\"operation\":\"StixCoreObjectsMultiHeatMapTimeSeriesQuery\",\"operation_query\":\"query StixCoreObjectsMultiHeatMapTimeSeriesQuery($operation:StatsOperation!$startDate:DateTime!$endDate:DateTime!$interval:String!$timeSeriesParameters:[StixCoreObjectsTimeSeriesParameters]){stixCoreObjectsMultiTimeSeries(operation:$operation startDate:$startDate endDate:$endDate interval:$interval timeSeriesParameters:$timeSeriesParameters){data{date value}}}\",\"size\":279,\"time\":5,\"timestamp\":\"2023-03-03T14:58:45.850Z\",\"type\":\"READ_ERROR\",\"user\":{\"ip\":\"XXXX\",\"referer\":\"XXXX/dashboard/workspaces/dashboards/afb9757f-ef39-41b3-90ae-cc89f8448d63\",\"user_id\":\"XXXX\"},\"variables\":{\"endDate\":\"2023-03-03T16:58:45+02:00\",\"interval\":\"day\",\"operation\":\"count\",\"startDate\":\"2022-03-03T00:00:00+02:00\",\"timeSeriesParameters\":[{\"field\":\"created\",\"filters\":[{\"filterMode\":\"or\",\"key\":\"relationship_type\",\"operator\":\"eq\",\"values\":[\"related-to\"]}],\"types\":[\"Malware\"]}]},\"version\":\"5.5.3\"}\n"
`
Filters which cause a problem:
`
      "timeSeriesParameters": [
        {
          "field": "created",
          "filters": [
            {
              "filterMode": "or",
              "key": "relationship_type",
              "operator": "eq",
              "values": [
                "related-to"
              ]
            }
          ],
          "types": [
            "Malware"
          ]
        }
      ]
`

## Expected Output

Expect the visualization to be added to the dashboard instead.

## Actual Output

The dashboard becomes inaccessible due to the error.

### PR DESCRIPTION
**Pull request by richard-julien**
Info in https://github.com/OpenCTI-Platform/opencti/issues/2921

### CODE REVIEW

File: opencti-platform/opencti-front/src/private/components/workspaces/WorkspaceHeader.js [hunk 1]
```diff
@@ -75,77 +74,54 @@ const workspaceMutation = graphql`
...
+  const [openTag, setOpenTag] = useState(false);
+  const [openTags, setOpenTags] = useState(false);
 
-  onSubmitCreateTag(element, data, { resetForm }) {
-    const currentTags = this.getCurrentTags();
-    if (
-      (currentTags === null || !currentTags.includes(data.new_tag))
-      && data.new_tag !== ''
-    ) {
+  const handleToggleOpenTags = () => setOpenTags(!openTags);
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/components/workspaces/WorkspaceHeader.js]: Not the best practice to do it that way, you should do setOpenTags((current) => !current)

File: opencti-platform/opencti-front/src/private/components/workspaces/WorkspaceHeader.js [hunk 2]
```diff
@@ -75,77 +74,54 @@ const workspaceMutation = graphql`
...
+  const handleToggleCreateTag = () => setOpenTag(!openTag);
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/components/workspaces/WorkspaceHeader.js]: Same

File: opencti-platform/opencti-front/src/private/components/workspaces/WorkspaceHeader.js [hunk 3]
```diff
@@ -75,77 +74,54 @@ const workspaceMutation = graphql`
...
+  const getCurrentTags = () => workspace.tags;
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/components/workspaces/WorkspaceHeader.js:88]: Why is this useful ?

Comment (richard-julien) [reply opencti-platform/opencti-front/src/private/components/workspaces/WorkspaceHeader.js:88]: Stupid basic pure function conversion :) Will review that

File: opencti-platform/opencti-front/src/private/components/workspaces/WorkspaceHeader.js (hunk 1 above)
Comment (richard-julien) [reply opencti-platform/opencti-front/src/private/components/workspaces/WorkspaceHeader.js]: Hum can you explain why?

Comment (Kedae) [reply opencti-platform/opencti-front/src/private/components/workspaces/WorkspaceHeader.js]: It's a preconization from react https://reactjs.org/docs/faq-state.html#how-do-i-update-state-with-values-that-depend-on-the-current-state

Comment (richard-julien) [reply opencti-platform/opencti-front/src/private/components/workspaces/WorkspaceHeader.js]: This is also true for hook? I donk think useState have the same requirement.

Comment (Kedae) [reply opencti-platform/opencti-front/src/private/components/workspaces/WorkspaceHeader.js]: That's actually a good question never thought of that. I've always kept that practice in my code. If you don't get weird behavior we may assume that's no longer needed.

**Review by SamuelHassine — APPROVED** [review]

**Pull request merged**